src/main/java/org/springframework/data/r2dbc/dialect/IndexedBindMarkers.java

```diff
@@ -0,0 +1,87 @@
+package org.springframework.data.r2dbc.dialect;
+
+import io.r2dbc.spi.Statement;
+
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+/**
+ * Index-based bind marker. This implementation creates indexed bind markers using a numeric index and an optional
+ * prefix for bind markers to be represented within the query string.
+ *
+ * @author Mark Paluch
+ */
+class IndexedBindMarkers implements BindMarkers {
```

Comment:
I think the names `IndexedBindM...` and `NamedBindM...` are misleading since they really are both based on an index getting increment. 

Maybe `SimpleIndexed...` and `HintHonouringIndexed...` might be better choices.

Reply:
The name is about how binding is supposed to work. Indexed binding uses integer indexes (`bind(int, …)`) to bind values to a `Statement` whereas named bind markers uses identifiers (`bind(Object, …)`) to bind values.